### ISSUE TITLE
Table: Line graph where... does not work properly if nrow < ncol

### ISSUE THREAD

**Issue by jjatria** [praat/praat#34]
The command `Line graph where...` for Tables will not work if the column that is set to be the "vertical column" is the final one (=rightmost) in the Table object, **and** the Table object has fewer rows than columns.

This script exemplifies the problem:

~~~
# Generate sample Tables
columns$ = "a b c d e f g h i j k l m n o p q r s t u v w x y z"
Erase all
for rows from 3 to 5
  for cols from 3 to 5
    table = Create Table with column names: if rows < cols then "bad" else "good" fi,
      ... rows, left$(columns$, (2 * cols) - 1)

    # Populate Table
    for y to rows
      for x to cols
        Set numeric value: y, mid$(columns$, (2 * x) - 1), y
      endfor
    endfor

    # Problem only exists when drawing the final (=rightmost) column
    last_column$ = mid$(columns$, (2 * cols) - 1)

    # The problematic command
    Line graph where: last_column$, 0, 0, "", 0, 0, "+", 0, "yes", "1; (= everything)"

    # Problem only exists with Tables where nrow < ncol
    pauseScript: "Did ",
      ... if rows < cols then "not " else "" fi,
      ... "draw (", rows, ", ", cols, ")"
    removeObject: table
    Erase all
  endfor
endfor
~~~

The problem exists up to and including 5.4.19, in Linux and Mac at least.

**Comment by jjatria** [comment]
The problem persists in 5.4.21

**Comment by davidweenink** [comment]
The problem will be solved in 5.4.22.

**Comment by davidweenink** [comment]
opgelost zie github.

David

________________________________________

**Comment by PaulBoersma** [comment]
hoi David,

ja kunt de issue dan sluiten op GitHub.

En voor de volgende keer (dit leerde ik gisteren): je had "#34" kunnen vermelden in je commit; schijnt gewoonte te zijn, want mensen kunnen dan bij de closed issues doorklikken naar de manier waarop de bug verwijderd is.

groetjes,
Paul

> Op 8 okt. 2015, om 16:16 heeft davidweenink <notifications@github.com> het volgende geschreven:
> 
> opgelost zie github.
> 
> David
> 
> ________________________________________
> From: Paul Boersma [notifications@github.com]
> Sent: Wednesday, October 07, 2015 21:44
> To: praat/praat
> Cc: Weenink, David
> Subject: Re: [praat] Table: Line graph where... does not work properly if nrow < ncol (#34)
> 
> Assigned #34<https://github.com/praat/praat/issues/34> to @davidweenink<https://github.com/davidweenink>.
> 
> �
> Reply to this email directly or view it on GitHub<https://github.com/praat/praat/issues/34#event-429540489>.
> 
> —
> Reply to this email directly or view it on GitHub.
> 

_____

Paul Boersma
University of Amsterdam
Linguistics
Spuistraat 210, room 303
1012VT Amsterdam, The Netherlands
http://www.fon.hum.uva.nl/paul/

**Comment by jjatria** [comment]
Confirmed as fixed in 5.4.22.
Thanks!
